htdocs/composer.json

```diff
@@ -0,0 +1,85 @@
+{
+	"name": "impresscms/impresscms",
+	"description": "ImpressCMS - A dynamic and user-friendly Content Management System",
+	"type": "project",
+	"keywords": ["cms", "content-management", "php", "mysql", "web"],
+	"homepage": "https://www.impresscms.org/",
+	"license": "GPL-2.0-or-later",
+	"authors": [
+		{
+			"name": "The ImpressCMS Project",
+			"homepage": "https://www.impresscms.org/",
+			"role": "Developer"
+		}
+	],
+	"require": {
+		"php": ">=7.4.0",
+		"ext-gd": "*",
+		"ext-json": "*",
+		"ext-mbstring": "*",
+		"ext-mysqli": "*",
+		"ext-pcre": "*",
+		"ext-pdo": "*",
+		"ext-session": "*",
+		"ext-xml": "*",
+		"ext-zlib": "*",
+		"composer/composer": "2.8.12",
+		"ezyang/htmlpurifier": "^4.19",
+		"simplepie/simplepie": "^1.9",
+		"smottt/wideimage": "^1.1"
+	},
+	"require-dev": {
+		"phpunit/phpunit": "^9.0",
+		"squizlabs/php_codesniffer": "^3.0"
+	},
+	"autoload": {
+		"psr-4": {
+			"Icms\\": "libraries/icms/"
+		},
+		"psr-0": {
+			"icms_": "libraries/"
+		},
+		"classmap": [
+			"libraries/icms.php"
+		]
+	},
+	"autoload-dev": {
+		"psr-4": {
+			"Icms\\Tests\\": "../tests/"
+		}
+	},
+	"config": {
+		"optimize-autoloader": true,
+		"classmap-authoritative": false,
+		"apcu-autoloader": true,
+		"sort-packages": true,
+		"allow-plugins": {
+			"*": false
+		},
+		"cache-dir": "var/cache/composer",
+		"vendor-dir": "vendor"
+	},
+	"scripts": {
+		"test": "phpunit",
+		"cs-check": "phpcs --standard=PSR12 libraries/icms/",
+		"cs-fix": "phpcbf --standard=PSR12 libraries/icms/",
```

Comment:
i prefer config file because in icms there are more paths files then libraries/icms/ that should be checked... also I think this is separate PR - don't mix responsibilities!